pkg/registry/apis/provisioning/resources/parser.go

```diff
@@ -226,12 +231,6 @@ func (f *ParsedResource) DryRun(ctx context.Context) error {
 			DryRun: []string{"All"},
 		})
 	}
-
-	// When the name is missing (and generateName is configured) use the value from DryRun
-	if f.Obj.GetName() == "" && f.DryRunResponse != nil {
-		f.Obj.SetName(f.DryRunResponse.GetName())
```

Comment:
this logic did not save a value with the generated name!  because it runs *after* we saved to git/file